browser/ai_chat/content_agent_tool_provider.cc

```diff
@@ -65,7 +65,11 @@ ContentAgentToolProvider::ContentAgentToolProvider(
   CreateTools();
 }
 
-ContentAgentToolProvider::~ContentAgentToolProvider() = default;
+ContentAgentToolProvider::~ContentAgentToolProvider() {
+  // When this tool provider and its owned uses (e.g. conversation)are closed,
+  // we should hand back state of the tab to regular uses of the browser.
+  StopAllTasks();
```

Comment:
Just double checking, but is this okay when the Profile is destroyed? I think it should be because we destroy conversations at `AIChatService::Shutdown` rather than `AIChatService::~AIChatService`, but wanted to flag it just in case :smile: